test/integration/targets/debug/args_templating.yml

```diff
@@ -0,0 +1,95 @@
+- name: splat args that don't resolve to a dict
+  debug:
+  args: '{{ "i am a string go bang" }}'
+  ignore_errors: true
+  register: result
+
+- assert:
+    that:
+    - result is failed
+    - result.msg is contains "Finalization of task args"
+
+- name: merge splatted and local args with omit (local takes precedence for conflicts)
+  vars:
+    should_not_see_me: '{{ 1 / 0 }}'
+    some_var_value: some var value
+    splatted_debug_args:
+      msg: '{{ omit }}'   # this should be dropped before argspec validation and not cause a failure
+      var: should_not_see_me  # this should be masked by the explicit on-task `var` arg value
+  debug:
+    var: some_var_value
+  args: '{{ splatted_debug_args }}'
+  register: result
+
+- assert:
+    that:
+    - result.some_var_value == "some var value"
+
+- name: verify that local args are merged
+  vars:
+    debug_args:
+      var: this should conflict and fail
+  debug:
+    msg: this one please
+  args: '{{ debug_args }}'
+  ignore_errors: true
+  register: result
+
+- assert:
+    that:
+    - result is failed
+    - '"mutually exclusive" in result.msg'
+
+- name: entire args dict omitted
+  debug:
+  args: '{{ omit }}'
+  register: result
+
+- assert:
+    that: result.msg == "Hello world!"
+
+- name: templated var input can be omitted
+  debug:
+    var: '{{ omit }}'
+  register: result
+
+- assert:
+    that:
+      - result.msg == "Hello world!"
+
+- name: static var that resolves to omit should give a warning and placeholder output
+  vars:
+    indirect_omit: '{{ omit }}'
+  debug:
+    var: indirect_omit
+  register: result
+
+- assert:
+    that:
+      - result.indirect_omit == "<<Omit>>"
+      - result.warnings | length == 1
+      - result.warnings[0] is contains "could not be omitted"
+
+- name: templated verbosity should work
+  debug:
+    verbosity: '{{ 100 }}'
+  register: result
+
+- assert:
+    that:
+      - result is skipped
+
+# DTFIX-RELEASE: this capability works in devel; we'd need to have a special templating mode that propagates trust for all
+# inline constant strings in visit_Const (vs just those with handlebars today)... If present, this allows some degree
+# of dynamic variable selection for `var` (eg ternary), but you could also just use `msg`.
+- name: templated var with trusted scalar works
+  vars:
+    trusted_varname: hi mom
+  debug:
+    var: '{{ "trusted_varname" }}'
+  register: result
+
+- assert:
+    that:
+      #- result.trusted_varname == "hi mom"  # this is what devel does
+      - result.trusted_varname == "trusted_varname"
```

Comment:
```suggestion
      - result.trusted_varname == "trusted_varname"
      
- name: test templating
  debug:
     var: "{{ foo | default(True) }}"
```

Reply:
This test passes in devel

Reply:
The change in behavior is intentional. In devel that task produces:

```
ok: [localhost] => {
    "true": "VARIABLE IS NOT DEFINED!: "
}
```

In this PR it produces:

```
[ERROR]: Task failed: Action failed: argument 'var' is of type bool and we were unable to convert to _check_type_str_no_conversion: 'True' is not a string and conversion is not allowed
Origin: /Users/mclay/code/mattclay/ansible-private/demo.yml:4:7

2   gather_facts: no
3   tasks:
4     - name: test templating
        ^ column 7

fatal: [localhost]: FAILED! => {"msg": "argument 'var' is of type bool and we were unable to convert to _check_type_str_no_conversion: 'True' is not a string and conversion is not allowed"}
```

Reply:
The error message still confuses `argument 'var' is of type bool`. Feels like `var` accepts `bool` only and we provided something else.